torch/csrc/jit/tensorexpr/eval.h

```diff
@@ -574,7 +574,19 @@ class SimpleIREvaluator : public CodeGen, public IRVisitor {
 
   TORCH_API void visit(const IfThenElse* v) override {
     v->condition()->accept(this);
-    if (value_.as<int>()) {
+    bool cond_v;
+    switch (value_.dtype().scalar_type()) {
+#define TYPE_CASE(Type, Name)   \
+  case ScalarType::Name: {      \
+    cond_v = value_.as<Type>(); \
+  } break;
+      AT_FORALL_SCALAR_TYPES_AND(Bool, TYPE_CASE);
```

Comment:
Not supporting Half? or just doesn't support coerce to bool?

Maybe a comment here either way.

Reply:
I added an explicit case to throw a more verbose error for `Half`.